pkg/auth/server/errorpage/errorpage_test.go

```diff
@@ -0,0 +1,154 @@
+
+		{
+			resp := httptest.NewRecorder()
+			handled, err := handler.AuthenticationError(nil, resp, &http.Request{Header: testCase.Headers})
+			if err != nil {
+				t.Errorf("%s: unexpected error: %#v", k, err)
+				continue
+			}
+			if testCase.ExpectHandled != handled {
+				t.Errorf("%s: unexpected value for 'handled': %#v", k, handled)
+				continue
+			}
+		}
+
+		{
+			resp := httptest.NewRecorder()
+			handled, err := handler.GrantError(nil, resp, &http.Request{Header: testCase.Headers})
+			if err != nil {
+				t.Errorf("%s: unexpected error: %#v", k, err)
+				continue
+			}
+			if testCase.ExpectHandled != handled {
+				t.Errorf("%s: unexpected value for 'handled': %#v", k, handled)
+				continue
+			}
+		}
+	}
+}
+
+func TestValidateErrorPageTemplate(t *testing.T) {
+	testCases := map[string]struct {
+		Template      string
+		TemplateValid bool
+	}{
+		"broken template": {
+			Template:      `Test {{ .BadField }}`,
+			TemplateValid: false,
+		},
+		"default template": {
+			Template:      defaultErrorPageTemplateString,
+			TemplateValid: true,
+		},
+		"template example": {
+			Template:      ErrorPageTemplateExample,
+			TemplateValid: true,
+		},
+		"original template example": {
+			Template:      originalErrorPageTemplateExample,
+			TemplateValid: true,
+		},
+	}
+
+	for k, testCase := range testCases {
+		allErrs := ValidateErrorPageTemplate([]byte(testCase.Template))
+		if testCase.TemplateValid {
+			for _, err := range allErrs {
+				t.Errorf("%s: template validation failed when it should have succeeded: %v", k, err)
+			}
+		} else if len(allErrs) == 0 {
+			t.Errorf("%s: template validation succeeded when it should have failed", k)
+		}
+	}
+}
+
+// Make sure the original version of the default template always validates
+// this is to avoid breaking existing customized templates.
+const originalErrorPageTemplateExample = `<!DOCTYPE html>
+<!--
+
+This template can be modified and used to customize the error page. To replace
+the error page, set master configuration option oauthConfig.templates.error to
+the path of the template file.
+
+oauthConfig:
+  templates:
+    error: templates/error-template.html
+
+The Error field contains an error message, which is human readable, and subject to change.
+Default error messages are intentionally generic to avoid leaking information about authentication errors.
+
+The ErrorCode field contains a programmatic error code, which may be (but is not limited to):
+- mapping_claim_error
+- mapping_lookup_error
+- authentication_error
+- grant_error
+-->
+<html>
+  <head>
+    <title>Error</title>
+    <style type="text/css">
+      body {
+        font-family: "Open Sans", Helvetica, Arial, sans-serif;
+        font-size: 14px;
+        margin: 15px;
+      }
+    </style>
+  </head>
+  <body>
+
+    <div>
+		{{ if eq .ErrorCode "authentication_error" }}
+			An authentication error occurred.
+		{{ else if eq .ErrorCode "mapping_claim_error"   }}
```

Comment:
Extraneous spaces near your braces here and below, and the indentation is inconsistent